libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -90,348 +116,594 @@ string ASTJsonConverter::sourceLocationToString(SourceLocation const& _location)
+		make_pair("argumentTypes", typePointerToJson(_node.annotation().argumentTypes))
+	});
+	return false;
 }
 
 bool ASTJsonConverter::visit(MemberAccess const& _node)
 {
-	addJsonNode(_node, "MemberAccess", {
-		make_pair("member_name", _node.memberName()),
-		make_pair("type", type(_node))
-	}, true);
-	return true;
+	setJsonNode(_node, "MemberAccess", {
+		make_pair("memberName", _node.memberName()),
+		make_pair("expression", toJson(_node.expression())),
+		make_pair("referencedDeclaration", idOrNull(_node.annotation().referencedDeclaration)),
+		make_pair("isConstant", _node.annotation().isConstant),
+		make_pair("isPure", _node.annotation().isPure),
+		make_pair("isLValue", _node.annotation().isLValue),
+		make_pair("lValueRequested", _node.annotation().lValueRequested),
+		make_pair("typeDescriptions", typePointerToJson(_node.annotation().type)),
+		make_pair("argumentTypes", typePointerToJson(_node.annotation().argumentTypes))
+	});
+	return false;
 }
 
 bool ASTJsonConverter::visit(IndexAccess const& _node)
 {
-	addJsonNode(_node, "IndexAccess", { make_pair("type", type(_node)) }, true);
-	return true;
+	setJsonNode(_node, "IndexAccess", {
+		make_pair("baseExpression", toJson(_node.baseExpression())),
+		make_pair("indexExpression", toJsonOrNull(_node.indexExpression())),
+		make_pair("isConstant", _node.annotation().isConstant),
+		make_pair("isPure", _node.annotation().isPure),
+		make_pair("isLValue", _node.annotation().isLValue),
+		make_pair("lValueRequested", _node.annotation().lValueRequested),
+		make_pair("typeDescriptions", typePointerToJson(_node.annotation().type)),
+		make_pair("argumentTypes", typePointerToJson(_node.annotation().argumentTypes))
+	});
+	return false;
 }
 
 bool ASTJsonConverter::visit(Identifier const& _node)
 {
-	addJsonNode(_node, "Identifier",
-				{ make_pair("value", _node.name()), make_pair("type", type(_node)) });
-	return true;
+	Json::Value overloads(Json::arrayValue);
+	for (auto const& dec: _node.annotation().overloadedDeclarations)
+		overloads.append(Json::Value(nodeId(*dec)));
+	setJsonNode(_node, "Identifier", {
+		make_pair("value", _node.name()),
+		make_pair("referencedDeclaration", idOrNull(_node.annotation().referencedDeclaration)),
+		make_pair("overloadedDeclarations", overloads),
+		make_pair("typeDescriptions", typePointerToJson(_node.annotation().type)),
+		make_pair("argumentTypes", typePointerToJson(_node.annotation().argumentTypes))
+	});
+	return false;
 }
 
 bool ASTJsonConverter::visit(ElementaryTypeNameExpression const& _node)
 {
-	addJsonNode(_node, "ElementaryTypeNameExpression", {
+	setJsonNode(_node, "ElementaryTypeNameExpression", {
 		make_pair("value", _node.typeName().toString()),
-		make_pair("type", type(_node))
+		make_pair("isConstant", _node.annotation().isConstant),
+		make_pair("isPure", _node.annotation().isPure),
+		make_pair("isLValue", _node.annotation().isLValue),
+		make_pair("lValueRequested", _node.annotation().lValueRequested),
```

Comment:
As a generic question: shouldn't we take the opportunity to review the naming of these fields before it becomes part of the "public API" via the AST?

Some of this seems less consistent, e.g. `isLValue` vs. `lValueRequested`. I understand the goal is to have first letter lowercase.